html/captive-portal/lib/captiveportal/PacketFence/Controller/Oauth2.pm

```diff
@@ -88,11 +89,19 @@ sub oauth2_client {
     } elsif (lc($provider) eq 'windowslive'){
         $type = pf::Authentication::Source::WindowsLiveSource->meta->get_attribute('type')->default;
         $token_scheme = "auth-header:Bearer";
+    } elsif (lc($provider) eq 'twitter'){
+        $type = pf::Authentication::Source::TwitterSource->meta->get_attribute('type')->default;
     }
 
     if ($type) {
         my $source = $portalSession->profile->getSourceByType($type);
         if ($source) {
+            # Twitter source is special, we need our homemade lib
+            # that's included in the source
+            if($type eq "Twitter"){
```

Comment:
Put a space before and after a condtion and use single quotes for constant strings:

```perl
 if ($type eq 'Twitter') {
```